### ISSUE TITLE
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference

### ISSUE THREAD

**Issue by RicardAparicio** [OneSignal/OneSignal-Android-SDK#1261]
**Description:**
Our app is crashing almost 1% in production since last release 4.1.0. As far as we can see, the line producing the crash is added in this version, all the crashes are happening in background.

**Environment**
minSdkVersion 21 
compileSdkVersion 29

**Steps to Reproduce Issue:**
Could not reproduce the crash yet.

**Stacktrace:**

```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
       at com.onesignal.WebViewManager.stopped(WebViewManager.java:320)
       at com.onesignal.ActivityLifecycleHandler.onActivityStopped(ActivityLifecycleHandler.java:110)
       at com.onesignal.ActivityLifecycleListener.onActivityStopped(ActivityLifecycleListener.java:103)
       at android.app.Application.dispatchActivityStopped(Application.java:521)
       at android.app.Activity.dispatchActivityStopped(Activity.java:1362)
       at android.app.Activity.onStop(Activity.java:2515)
       at androidx.fragment.app.FragmentActivity.onStop(FragmentActivity.java:490)
       at androidx.appcompat.app.AppCompatActivity.onStop(AppCompatActivity.java:216)
       at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1473)
       at android.app.Activity.performStop(Activity.java:8143)
       at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:4700)
       at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:4678)
       at android.app.ActivityThread.handleStopActivity(ActivityThread.java:4753)
       at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:233)
       at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2067)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:224)
       at android.app.ActivityThread.main(ActivityThread.java:7519)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995)
```

**Comment by beetlebum** [comment]
We're getting hundreds of these crashes every time we send out an In App Message. We're not doing anything special with the IAM, no triggers or anything like that, it's all set up in the dashboard.

```java.lang.RuntimeException: 
  at android.app.ActivityThread.callActivityOnStop (ActivityThread.java:4755)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:4725)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:4800)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:233)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:173)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2123)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7710)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:516)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:950)
Caused by: java.lang.NullPointerException: 
  at com.onesignal.WebViewManager.stopped (WebViewManager.java:320)
  at com.onesignal.ActivityLifecycleHandler.onActivityStopped (ActivityLifecycleHandler.java:110)
  at com.onesignal.ActivityLifecycleListener.onActivityStopped (ActivityLifecycleListener.java:103)
  at android.app.Application.dispatchActivityStopped (Application.java:535)
  at android.app.Activity.dispatchActivityStopped (Activity.java:1331)
  at android.app.Activity.onStop (Activity.java:2464)
  at androidx.fragment.app.FragmentActivity.onStop (FragmentActivity.java:558)
  at androidx.appcompat.app.AppCompatActivity.onStop (AppCompatActivity.java:216)
  at android.app.Instrumentation.callActivityOnStop (Instrumentation.java:1591)
  at android.app.Activity.performStop (Activity.java:8036)
  at android.app.ActivityThread.callActivityOnStop (ActivityThread.java:4747)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:4725)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:4800)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:233)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:173)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2123)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7710)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:516)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:950)```

![image](https://user-images.githubusercontent.com/227258/111868367-73ad9e80-8971-11eb-8213-9d107a01dd72.png)

**Comment by KristapsKrumins** [comment]
Same here with version 4.2.0

![image](https://user-images.githubusercontent.com/16702487/112600808-4f8f0900-8e1a-11eb-9632-ed9590275b79.png)

Models:
-Galaxy S21 Ultra 5G with Android version 11
-HUAWEI P30 lite with Android version 10
-Galaxy A40 with Android version 10
-Galaxy A20s with Android version 10
-Galaxy Note8 with Android version 8

**Comment by vantangst** [comment]
Same here with latest version(i guess) - implementation 'com.onesignal:OneSignal:[4.0.0, 4.99.99]'

<img width="949" alt="image" src="https://user-images.githubusercontent.com/13620155/114700330-99994980-9d4b-11eb-93c4-2bdba075bf90.png">

**Comment by ZakariaEsbaiy** [comment]
**WebViewManager.java line 2**
**com.onesignal.WebViewManager.stopped**

**This issue has 37 crash events affecting 37 users in the last 24 hours**

**Stack trace**

**_Fatal Exception: java.lang.RuntimeException: Unable to stop activity {info.szmdev.simplenumberfind/info.szmdev.simplenumberfind.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference_**
       at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5294)
       at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:5259)
       at android.app.ActivityThread.handleStopActivity(ActivityThread.java:5339)
       at android.app.servertransaction.StopActivityItem.execute(StopActivityItem.java:40)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2277)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8425)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:596)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)


**_Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference_**
       at com.onesignal.WebViewManager.stopped(WebViewManager.java:2)
       at com.onesignal.ActivityLifecycleHandler.onActivityStopped(ActivityLifecycleHandler.java:3)
       at com.onesignal.ActivityLifecycleListener.onActivityStopped(ActivityLifecycleListener.java:3)
       at android.app.Application.dispatchActivityStopped(Application.java:515)
       at android.app.Activity.dispatchActivityStopped(Activity.java:1477)
       at android.app.Activity.onStop(Activity.java:2641)
       at androidx.fragment.app.FragmentActivity.markFragmentsCreated(FragmentActivity.java)
       at androidx.appcompat.app.AppCompatActivity.onStop(AppCompatActivity.java)
       at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1476)
       at android.app.Activity.performStop(Activity.java:8414)
       at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5286)
       at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:5259)
       at android.app.ActivityThread.handleStopActivity(ActivityThread.java:5339)
       at android.app.servertransaction.StopActivityItem.execute(StopActivityItem.java:40)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2277)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8425)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:596)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)


![screencapture-console-firebase-google-u-0-project-callerid-project-crashlytics-app-android-info-szmdev-simplenumberfind-issues-f09844c468cf9367e054c1496dc218f3-2021-04-22-02_19_42](https://user-images.githubusercontent.com/10187373/115646252-59b90000-a311-11eb-9784-a478009ba262.png)

**Comment by KristapsKrumins** [comment]
The issue seems to be fixed with version 4.3.4.

**Comment by Jeasmine** [comment]
This error has been confirmed to be fixed with https://github.com/OneSignal/OneSignal-Android-SDK/pull/1304
